anta/tests/routing/bgp.py

```diff
@@ -1628,3 +1632,93 @@ def test(self) -> None:
             self.result.is_success()
         else:
             self.result.is_failure(f"The following BGP peer(s) are not configured or maximum routes and maximum routes warning limit is not correct:\n{failures}")
+
+
+class VerifyBGPRouteOrigin(AntaTest):
+    """Verifies BGP route origin for the provided IPv4 Network(s).
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if the BGP route's origin matches expected origin type and next-hop address.
+    * Failure: The test will fail if the BGP route's origin does not matches with expected origin type, next-hop address or BGP route entry(s) not found.
+
+    Examples
+    --------
+    ```yaml
+    anta.tests.routing:
+      bgp:
+        - VerifyBGPRouteOrigin:
+            route_entries:
+                - prefix: 10.100.0.128/31
+                  vrf: default
+                  route_paths:
+                    - nexthop: 10.100.0.10
+                      origin: Igp
+                    - nexthop: 10.100.4.5
+                      origin: Incomplete
+    ```
+    """
+
+    name = "VerifyBGPRouteOrigin"
+    description = "Verifies BGP route origin for the provided IPv4 Network(s)."
+    categories: ClassVar[list[str]] = ["bgp"]
+    commands: ClassVar[list[AntaCommand | AntaTemplate]] = [AntaTemplate(template="show ip bgp {prefix} vrf {vrf}", revision=3)]
+
+    class Input(AntaTest.Input):
+        """Input model for the VerifyBGPRouteOrigin test."""
+
+        route_entries: list[BgpRoute]
+        """List of BGP route(s)"""
+
+        class BgpRoute(BaseModel):
+            """Model for a BGP route(s)."""
+
+            prefix: IPv4Network
+            """IPv4 network address"""
+            vrf: str = "default"
+            """Optional VRF. If not provided, it defaults to `default`."""
+            route_paths: list[dict[str, IPv4Address | Literal["Igp", "Egp", "Incomplete"]]]
+            """A list of dictionaries represents a BGP path.
+            - `nexthop`: The next-hop IP address for the path.
+            - `origin`: The origin of the route."""
```

Comment:
could this be modeled properly?

Reply:
Updated input model for BGP route paths.
Thanks!!